internal/server/GameLoop.go

```diff
@@ -3,30 +3,115 @@ package server
 import (
 	"SOMAS2023/internal/common/objects"
 	"SOMAS2023/internal/common/physics"
+	"SOMAS2023/internal/common/utils"
 	"fmt"
+
+	"github.com/google/uuid"
 )
 
 func (s *Server) RunGameLoop() {
-	// Each agent makes a decision
+
+	// -------------------------- PROCESS JOINING REQUESTS -------------------------
+	// 1. group agents that have onBike = false by the bike they are trying to join
+	bikeRequests := s.GetJoiningRequests()
+	// 2. pass to agents on each of the desired bikes a list of all agents trying to join
+	for bike, pendingAgents := range bikeRequests {
+		agents := s.megaBikes[bike].GetAgents()
+
+		responses := make([](map[uuid.UUID]bool), 0) // list containing all the agents' ranking
+		for _, agent := range agents {
+			responses = append(responses, agent.DecideJoining(pendingAgents))
+		}
+		// 3. accept agents based on the response outcome (it will have to be a ranking system, as only 8-n bikers can be accepted)
+		acceptedRanked := GetAcceptanceRanking(responses)
+		for _, accepted := range acceptedRanked[:(8 - len(s.megaBikes[bike].GetAgents()))] {
+			s.GetAgentMap()[accepted].ToggleOnBike()
+			s.SetBikerBike(s.GetAgentMap()[accepted], bike)
```

Comment:
This doesn't seem right. Surely you'd want to iterate over all of the agents - the first N of which can be put on the bike (no toggle needed), and the rest just get put in 'limbo', and you DO toggle the bike status?

Reply:
We had thought of having the acceptance process happen at the beginning of the next stage to associate a "cost" with leaving a bike (that is, one round spent in the limbo), so all agents go in the limbo when they leave the bike and their request is processed at the beginning of next round
But I will bring up with the rest of the infrastructure team to see if we can change this in this iteration